packages/monaco/src/browser/textmate/textmate-tokenizer.ts

```diff
@@ -16,30 +16,55 @@
 
 import { INITIAL, StackElement, IGrammar } from 'monaco-textmate';
 
-export class State implements monaco.languages.IState {
+export class TokenizerState implements monaco.languages.IState {
 
     constructor(
-        public ruleStack: StackElement
+        public readonly ruleStack: StackElement
     ) { }
 
     clone(): monaco.languages.IState {
-        return new State(this.ruleStack);
+        return new TokenizerState(this.ruleStack);
     }
 
     equals(other: monaco.languages.IState): boolean {
-        return other &&
-            (other instanceof State) &&
-            (other === this || other.ruleStack === this.ruleStack)
-            ;
+        return other instanceof TokenizerState && (other === this || other.ruleStack === this.ruleStack);
     }
 
 }
 
-export function createTextmateTokenizer(grammar: IGrammar): monaco.languages.TokensProvider {
+/**
+ * Options for the TextMate tokenizer.
+ */
+export interface TokenizerOption {
+
+    /**
+     * Maximum line length that will be handled by the TextMate tokenizer. If the length of the actual line exceeds this
+     * limit, the tokenizer terminates and the tokenization of any subsequent lines might be broken.
+     *
+     * If the `lineLimit` is `false` it means, there are no line length limits. If the `lineLimit` is a number, then it must
+     * be a positive integer. Otherwise, an error will be thrown.
+     */
+    readonly lineLimit: number | false;
```

Comment:
Why not `lineLimit?: number` where `undefined` means no limit?

Reply:
We can do that too. I wanted to clearly separate the behavior.

Reply:
If I leave it out, how can I decide whether I have to fall back to the default `400` character or disable it at all?